src/modules/history/pages/SyncPage.tsx

```diff
@@ -0,0 +1,475 @@
+import { HistoryOptionsList } from '../components/HistoryOptionsList';
+
+interface PageProps {
+	serviceId: StreamingServiceId | null;
+}
+
+type LastSyncData = Record<
+	StreamingServiceId,
+	{
+		lastSync: number;
+		lastSyncId: string;
+	}
+>;
+
+const lastSyncData = {} as LastSyncData;
+
+export const SyncPage: React.FC<PageProps> = (props: PageProps) => {
+	const { serviceId } = props;
+
+	const service = serviceId ? streamingServices[serviceId] : null;
+	const store = getSyncStore(serviceId);
+	const api = serviceId ? getApi(serviceId) : null;
+
+	if (service && !lastSyncData[service.id]) {
+		const serviceOptions = BrowserStorage.options.streamingServices[service.id];
+		lastSyncData[service.id] =
+			service.hasAutoSync && serviceOptions?.autoSync && serviceOptions.autoSyncDays > 0
+				? {
+						lastSync: serviceOptions.lastSync,
+						lastSyncId: serviceOptions.lastSyncId,
+				  }
+				: {
+						lastSync: 0,
+						lastSyncId: '',
+				  };
+	}
+
+	const [_, setSyncOptionsChanged] = useState({});
+	const [content, setContent] = useState({
+		isLoading: serviceId ? store.data.page === 0 : false,
+		itemsChanged: {},
+		visibleItemsChanged: {},
+		...(serviceId
+			? lastSyncData[serviceId]
+			: {
+					lastSync: 0,
+					lastSyncId: '',
+			  }),
+		continueLoading: false,
+	});
+
+	const loadPreviousPage = () => {
+		void store.goToPreviousPage().update();
+	};
+
+	const loadNextPage = () => {
+		if (!service || !serviceId) {
+			return;
+		}
+		if (store.data.itemsToLoad > 0 && !store.data.hasReachedEnd) {
+			setContent((prevContent) => ({
+				...prevContent,
+				isLoading: true,
+			}));
+			const { hasReachedLastSyncDate } = store.data;
+			if (hasReachedLastSyncDate) {
+				store.setData({ hasReachedLastSyncDate: false });
+			}
+			EventDispatcher.dispatch('REQUESTS_CANCEL', null, { key: 'default' })
+				.then(() => api?.loadHistory(store.data.itemsToLoad, content.lastSync, content.lastSyncId))
+				.then(() => {
+					if (hasReachedLastSyncDate) {
+						void store.update();
+					} else {
+						void store.goToNextPage().update();
+					}
+				})
+				.catch(() => {
+					// Do nothing
+				});
+		} else if (store.data.hasNextPage) {
+			void store.goToNextPage().update();
+		}
+	};
+
+	const onPreviousPageClick = () => {
+		loadPreviousPage();
+	};
+
+	const onNextPageClick = () => {
+		loadNextPage();
+	};
+
+	const onSyncClick = async () => {
+		const missingWatchedDate = store.data.selectedItems.some((item) => item.isMissingWatchedDate());
+		if (missingWatchedDate) {
+			return EventDispatcher.dispatch('DIALOG_SHOW', null, {
+				title: I18N.translate('cannotSync'),
+				message: I18N.translate('cannotSyncMissingWatchedDate'),
+			});
+		}
+		setContent((prevContent) => ({
+			...prevContent,
+			isLoading: true,
+		}));
+		await TraktSync.sync(store.data.selectedItems);
+		if (serviceId) {
+			const lastSync =
+				store.data.selectedItems[0].watchedAt?.unix() ?? Math.trunc(Date.now() / 1e3);
```

Comment:
While trying to recreate the other bug, I encountered another here. After clearing lastsync date I was not able to re-enable the autosync because lastSync date where never set.
At this point the `store.data.selectedItems` is an empty array, so it fails with `TypeError: Cannot read property 'watchedAt' of undefined`. The array is however not empty on the await statement on line 134 just above. My theory is that the sync updated the DOM, and thus resets the selectedItems. I think this bug is solved by creating a copy of the array before calling `TraktSync.sync(items)`

Reply:
Fixed